src/components/UniversalSearch/UniversalSearch.tsx

```diff
@@ -0,0 +1,164 @@
+import styled from '@emotion/styled';
+import { push } from 'connected-react-router';
+import useDebounce from 'lib/effects/useDebounce';
+import { IUniversalSearcherResult } from 'lib/UniversalSearcher/UniversalSearcher';
+import * as React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import UniversalSearchInput from 'UniversalSearch/UniversalSearchInput';
+import { useUniversalSearch } from 'UniversalSearch/UniversalSearchProvider';
+import UniversalSearchSuggestionList from 'UniversalSearch/UniversalSearchSuggestionList';
+
+const UPDATE_DEBOUNCE_DELAY_MS = 250;
+
+const SearchWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 12px;
+  background-color: ${({ theme }) => theme.colors.darkBlueDarker3};
+  position: relative;
+  line-height: initial;
+`;
+
+interface IUniversalSearchProps extends React.ComponentPropsWithoutRef<'div'> {}
+
+const UniversalSearch: React.FC<IUniversalSearchProps> = React.memo(
+  ({ ...rest }) => {
+    const { searchTerm, search, searchResults, filters } = useUniversalSearch();
+    const debouncedSearchTerm: string = useDebounce(
+      searchTerm,
+      UPDATE_DEBOUNCE_DELAY_MS
+    );
+    const debouncedResults: IUniversalSearcherResult<unknown>[] = useDebounce(
```

Comment:
Is is really neccessary to debounce both input and output? Shouldn't search input be enough?

Reply:
There were a few problems with my logic: Searching wasn't actually debounced, just displaying the results was debounced. I fixed it now.

Search input doesn't need to be debounced for the search input (otherwise the contents of the input won'
t be updated with the correct text).

Actual searching should be always debounced, to prevent high resource load.

However, there are some checks related to showing or hiding the results dropdown, which need to use a debounced value of the search term. If we don't, then while typing, the user will see `No results found for your query`.